src/pineko/theory_card.py

```diff
@@ -22,3 +22,36 @@ def load(theory_id):
     with open(tcard_path, encoding="utf-8") as f:
         theory_card = yaml.safe_load(f)
     return theory_card
+
+def construct_assumption(tcard):
+    """This function return the appropriate assumption hash according to the scale Q0 of the fktable,
+    the matching scales of the heavy quarks and whether an intrinsic component of the charm is
+    allowed.
+
+    Parameters
+    ----------
+    tcard : dict
+        theory card
+
+    Returns
+    -------
+    str
+        assumption hash
+    """
+    # retrive the relevant info from theory card
+    Q0 = tcard['Q0']
+    match_scales = {'c' : tcard['kcThr']*tcard['mc'], 'b' : tcard['kbThr']*tcard['mb'] , 't' : tcard['ktThr']*tcard['mt'] }
+    ic = tcard["IC"]
+    hash = 'Nf'
+    act_flav = 6
+    mod = 'Ind'
+    if Q0 < match_scales['t']:
+        act_flav = 5
+    if Q0 < match_scales['b']:
+        act_flav = 4
+    if Q0 < match_scales['c']:
+        act_flav = 3
+        if ic:
+            act_flav += 1
```

Comment:
```suggestion
            act_flav += 1
            mod = 'Sym'
```

Reply:
> In any case, for this pass Nf4Ind and retain both charm contributions: neglecting c- will save very little memory, and it will be confusing the moment we decide to fit it. Do the consistent thing

It seemed to me that you wanted to avoid "Sym" even if it is what we are doing at the moment. Have I misunderstood?

Reply:
Slightly: I want to avoid `Sym` when we actually are in the 4 flavors (since it is inconsistent), and that is already what you are doing.

But if you see above in the list I told you that if you are parametrizing in the 3 flavors patch with IC (that is not what NNPDF is doing, at least for the time being), then you actually want `Sym`, because IC is unlocking only one degree of freedom (while the other would be behind something like ICA, that does not exist at the moment).

Reply:
Ah okok now it is clear. :)